src/library/dao/trackdao.cpp

```diff
@@ -264,7 +216,7 @@ void TrackDAO::saveTrack(Track* pTrack) {
         // a timestamp is used to keep track of when metadata has been
         // last synchronized. Exporting metadata will update this time
         // stamp on the track object!
-        if (m_pConfig && m_pConfig->getValueString(ConfigKey("[Library]","WriteAudioTags")).toInt() == 1) {
+        if (m_pConfig && m_pConfig->getValueString(ConfigKey("[Library]","SyncTrackMetadataExport")).toInt() == 1) {
```

Comment:
Syncing of metadata has two directions import and export. We are currently dealing with exporting the metadata from Mixxx, but there also is the use case of importing it automatically if the corresponding files has changed.

I already use an extended version that unconditionally imports the metadata from files upon each access with the corresponding option "SyncTrackMetadataImport". Currently it has some usability drawbacks that need to be resolved before publishing this feature, e.g. when selecting and accessing multiple tracks metadata from all files will be imported which might take some time.

Reply:
...and by using a new config key my personal settings don't get overwritten and lost each time I switch to master for testing ;) I forgot several times to re-enable the option and lost some edits. Mixxx currently sets and writes 0/False for this option.

Reply:
Ok, so we are heading to a two way sync 
Metadata feature, with only one parameter.
Is there a use case for a separate parameter?
I am afraid yes, because the user is able to switch, so we can easily create the situation that the Mixxx metadata is more recent than the file metadata and an implicite read from the file will omit the user entries from Mixxx

Reply:
No, two separate parameters are needed. Automatic import needs to be activated as a separate option and it will only be available if export of metadata is also enabled to avoid discarding modifications in Mixxx that have not been written into the corresponding file.

Safely importing metadata should also take the time stamp of the file into account and compare this to the time stamp of the last import/export in Mixxx. This feature is currently missing, we only have a simple flag. I left a TODO in the new class TrackRecord that introduces the concept.

Reply:
Ah OK, so we will have a "SyncTrackMetadataImport" in future. So the name makes sense.